ConsistencyManager/ConsistencyManager.swift

```diff
@@ -355,10 +363,17 @@ public class ConsistencyManager {
      It is called once on initialization to start the process. This will cause a cleanMemory call, but that's ok since it will be basically a no-op.
      */
     private func startGarbageCollection() {
-        // Weak here is necessay, otherwise, we'd have a retain cycle.
-        dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(UInt64(garbageCollectionInterval) * NSEC_PER_SEC)), dispatchQueue) { [weak self] in
-            self?.cleanMemory()
-            self?.startGarbageCollection()
+        // Saving a local variable so we only access this on the main thread
+        let garbageCollectionInterval = self.garbageCollectionInterval
+        // If garbageCollectionInterval is 0, this means it's disabled.
+        if garbageCollectionInterval > 0 {
+            // Weak here is necessay, otherwise, we'd have a retain cycle.
+            dispatch_after(dispatch_time(DISPATCH_TIME_NOW, Int64(UInt64(garbageCollectionInterval) * NSEC_PER_SEC)), dispatchQueue) { [weak self] in
```

Comment:
Yeah, I remember that. I'm not sure what was wrong with that. It looks like this has been working up to now so tempted to think this is ok.